lib/IRGen/AllocStackHoisting.cpp

```diff
@@ -57,6 +57,11 @@ static bool isHoistable(AllocStackInst *Inst, irgen::IRGenModule &Mod) {
   if (TI.isFixedSize())
     return false;
 
+  // Don't hoist weakly imported (weakly linked) types.
+  auto *nominalDecl = SILTy.getASTType()->getNominalOrBoundGenericNominal();
```

Comment:
What if you have a structural type, eg a function or tuple, containing a weak nominal? I think you need to do a findIf() to walk the type recursively instead

Reply:
(Functions are fixed size, so a tuple or optional is a better example)